JitsiMeetJS.ts

```diff
@@ -495,6 +509,104 @@ export default {
+            // Auto-initialize the library.
+            const autoInit = Symbol.for('kLJMAutoInit');
+
+            if (typeof window[autoInit] === 'undefined') {
+                window[autoInit] = true;
+
+                // TODO(saghul): review rtcstats initialization.
+                this.init({
+                    analytics: {
+                        rtcstatsEnabled: true,
+                        rtcstatsStoreLogs: true,
+                        rtcstatsSendSdp: true,
+                    },
+                    enableAnalyticsLogging: true,
+                });
+            }
+
+            const useStage = options.jaas?.useStaging ?? false;
+            const jaasDomain = useStage ? 'staging.8x8.vc' : '8x8.vc';
+            const opts = {
+                hosts: {
+                    domain: jaasDomain,
+                    muc: `conference.${appId}.${jaasDomain}`
+                },
+                conferenceRequestUrl: `https://${jaasDomain}/${appId}/conference-request/v1?room=${roomName}`,
+                serviceUrl: `wss://${jaasDomain}/${appId}/xmpp-websocket?room=${roomName}`,
+                websocketKeepAliveUrl: `https://${jaasDomain}/${appId}/_unlock?room=${roomName}`,
+                analytics: {
+                    rtcstatsEnabled: true,
+                    rtcstatsEndpoint: `wss://rtcstats-server-${useStage ? 'pilot' : '8x8'}.jitsi.net/`,
+                },
+            };
+
+            connectionOptions = merge(connectionOptions, opts);
+        }
+
+        const conn = new JitsiConnection(appId, token, connectionOptions);
+
+        function cleanupListeners() {
+            conn.removeEventListener(
+                JitsiConnectionEvents.CONNECTION_ESTABLISHED, onConnectionEstablished);
+            conn.removeEventListener(
+                JitsiConnectionEvents.CONNECTION_FAILED, onConnectionFailed);
+        }
+
+        function onConnectionEstablished() {
+            cleanupListeners();
+
+            const conf = conn.initJitsiConference(roomName, options.conferenceOptions ?? {});
+
+            d.resolve(conf);
+
+            // Make sure this runs after the promise was resolved so that local track events can be
+            // listened to.
+            queueMicrotask(() => {
+                for (const track of options.tracks || []) {
+                    conf.addTrack(track);
+                }
+
+                conf.join();
+
+                // TODO(saghul): set some sender / receiver constraints?
+            });
+        }
+
+        function onConnectionFailed(error: string) {
+            cleanupListeners();
+            d.reject(error);
+        }
+
+        conn.addEventListener(
+            JitsiConnectionEvents.CONNECTION_ESTABLISHED, onConnectionEstablished);
+        conn.addEventListener(
+            JitsiConnectionEvents.CONNECTION_FAILED, onConnectionFailed);
+        conn.connect({ name: roomName });
+
+        return d as unknown as Promise<JitsiConference>;
```

Comment:
What should we return here? Some options:

- Just return the conference and rely on garbage collection to kill the connection. Not a fan...
- Add a handler to close the connection when the conference ends. It would break breakout rooms, so not great either.
- Return an object with 2 properties: connection and conference, and perhaps a `destroy()` or `leave()` helper method that does The Right Thing. This might as well be the inception of `JitsiSession`!

Reply:
@jallamsetty1 thoughts on this?